src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs

```diff
@@ -86,5 +108,172 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
+                    //
+                    //   if (true)
+                    //     return;
+                    //   else          // <-- this else clause is two lines, but is not considered a multiline context
+                    //     return;
+                    //
+                    // ---
+                    // INTENTIONAL FALLTHROUGH
+                }
+                else
+                {
+                    var firstTokenAfterEmbeddedStatement = embeddedStatement.GetLastToken().GetNextToken();
+                    if (!FormattingRangeHelper.AreTwoTokensOnSameLine(firstTokenAfterEmbeddedStatement, statement.GetLastToken()))
+                    {
+                        // The part of the statement following the embedded statement does not fit on one line. Examples:
+                        //
+                        //   do
+                        //     SomeMethod();
+                        //   while (x < 0 ||    // <-- This condition is split across multiple lines.
+                        //          x > 10);
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines whether <paramref name="embeddedStatement"/> should use braces under a
+        /// <see cref="PreferBracesPreference.WhenMultiline"/> preference due to the presence of braces on one or more
+        /// sibling statements (the "context").
+        /// </summary>
+        private static bool RequiresBracesToMatchContext(SyntaxNode statement, StatementSyntax embeddedStatement)
+        {
+            if (!statement.IsKind(SyntaxKind.IfStatement, SyntaxKind.ElseClause))
+            {
+                // 'if' statements are the only statements that can have multiple embedded statements which are
+                // considered relative to each other.
+                return false;
+            }
+
+            var topLevelContext = GetTopLevelContext(statement);
+            if (AnySiblingUsesBraces(topLevelContext))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the top-most <see cref="IfStatementSyntax"/> for which <paramref name="ifStatementOrElseClause"/> is
+        /// part of the <c>if</c>/<c>else if</c>/<c>else</c> sequence.
+        /// </summary>
+        /// <remarks>
+        /// <para>For the purpose of brace usage analysis, the embedded statments of an <c>if</c>/<c>else if</c>/<c>else</c>
+        /// sequence are considered sibling statements, even though they don't appear as immediate siblings in the
+        /// syntax tree. This method walks up the syntax tree to find the <c>if</c> statement that starts the
+        /// sequence.</para>
+        /// </remarks>
+        private static IfStatementSyntax GetTopLevelContext(SyntaxNode ifStatementOrElseClause)
```

Comment:
`GetFirstIfStatementInChain` or `GetFirstIfStatementInIfStatementChain`

Reply:
➡️ Renamed to `GetOutermostIfStatementOfSequence`